README.md

```diff
@@ -46,7 +45,85 @@ Take notice that this project is build in a way that gives the developer a lot o
+1. Develop Mage targets in [magefile.go](magefile.go) and assosiated tasks in [.vscode/tasks.json](.vscode/tasks.json).
+
+Notable files:
+
+- [.devcontainer](.devcontainer/devcontainer.json) - Visual Studio Code Remote Cotainer configs
+- [.github](.github/workflows/build.yml) - GitHub Action workflow (CI build)
+- [.vscode](.vscode) - Visual Studio Code configs
+- [.golangci.yml](.golangci.yml) - GolangCI-Lint config
+- [docker-compose.yml](docker-compose.yml) - Compose file used in [CI Build](.github/workflows/build.yml)
+- [Dockerfile](Dockerfile) - Builder image used in [docker-compose.yml](docker-compose.yml) and [.devcontainer](.devcontainer/devcontainer.json)
+- [magefile.go](magefile.go) - Mage targets used in [docker-compose.yml](docker-compose.yml) and [.vscode/tasks.json](.vscode/tasks.json)
+
+## FAQ
+
+### Why Mage instead of Make
+
+Here is [why](https://github.com/magefile/mage#why).
+However, updating to Make is pretty straightforward.
+
+1. Replace [magefile.go](magefile.go) with a `Makefile` file:
+
+```make
+.DEFAULT_GOAL := help
+
+.PHONY: all
+all: ## full build: build, lint, test
+all: build lint test
+
+.PHONY: build
+build: ## go build
+	go build ./...
+
+.PHONY: lint
+lint: ## golangci-lint
+	golangci-lint run
+
+.PHONY: test
+test: ## go test with race detector and code covarage
+	go test -race -covermode=atomic
+
+.PHONY: help
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+```
+
+1. Remove Mage installation from [Dockerfile](Dockerfile).
+1. Update [docker-compose.yml](docker-compose.yml) and [.vscode/tasks.json](.vscode/tasks.json) to use `make`.
+
+If you want to use Make on bare-metal Windows, then you can:
+
+1. use [WSL (Windows Subsystem for Linux)](https://docs.microsoft.com/en-us/windows/wsl/install-win10) and [Visual Studio Code Remote SSH extension](https://code.visualstudio.com/docs/remote/wsl),
+1. install [Make Windows port to Git Bash](https://gist.github.com/evanwill/0207876c3243bbb6863e65ec5dc3f058).
+
+### Why nothing for GoLand
+
+The maintainer does not use GoLand. Fell free to create a pull request which should contain:
+
+- GoLand config to use golangci-lint,
+- GoLand config for Mage targets (like [.vscode/tasks.json](.vscode/tasks.json) for Visua Studio Code),
+- update [.gitignore](.gitignore) if required.
+- update README(README.md) if possible.
+
+### Why GitHub Actions, not any other CI server
+
+GitHub Actions is out-of-the-box if you are already using GitHub.
+However, changing to any other CI server should be very simple, becasue this repo uses Docker Compose to run CI build to make the transition easy.
+
+For [CircleCI](https://circleci.com/docs/2.0/executor-types/#using-machine) create `.circleci/config.yml` file:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    machine:
+      image: ubuntu-1604:201903-01
```

Comment:
Is it impossible to use `latest`?

Reply:
Does not work as you would think
https://discuss.circleci.com/t/default-machine-executor-image-update/29308
https://discuss.circleci.com/t/new-docker-versions-for-machine-images-on-circleci-2-0-on-2017-11-29/18144